client/rest-high-level/src/main/java/org/elasticsearch/client/indices/CreateIndexRequest.java

```diff
@@ -0,0 +1,394 @@
+     */
+    public CreateIndexRequest source(byte[] source, int offset, int length, XContentType xContentType) {
+        return source(new BytesArray(source, offset, length), xContentType);
+    }
+
+    /**
+     * Sets the settings and mappings as a single source.
+     *
+     * Note that the mapping definition should *not* be nested under a type name.
+     */
+    public CreateIndexRequest source(BytesReference source, XContentType xContentType) {
+        Objects.requireNonNull(xContentType);
+        source(XContentHelper.convertToMap(source, false, xContentType).v2(),
+            DeprecationHandler.THROW_UNSUPPORTED_OPERATION);
+        return this;
+    }
+
+    /**
+     * Sets the settings and mappings as a single source.
+     *
+     * Note that the mapping definition should *not* be nested under a type name.
+     */
+    @SuppressWarnings("unchecked")
+    public CreateIndexRequest source(Map<String, ?> source, DeprecationHandler deprecationHandler) {
+        for (Map.Entry<String, ?> entry : source.entrySet()) {
+            String name = entry.getKey();
+            if (SETTINGS.match(name, deprecationHandler)) {
+                settings((Map<String, Object>) entry.getValue());
+            } else if (MAPPINGS.match(name, deprecationHandler)) {
+                mapping((Map<String, Object>) entry.getValue());
+            } else if (ALIASES.match(name, deprecationHandler)) {
+                aliases((Map<String, Object>) entry.getValue());
+            }
+        }
+        return this;
+    }
+
+    public BytesReference mappings() {
+        return mappings;
+    }
+
+    public XContentType mappingsXContentType() {
+        return mappingsXContentType;
+    }
+
+    public Set<Alias> aliases() {
+        return this.aliases;
+    }
+
+    public ActiveShardCount waitForActiveShards() {
+        return waitForActiveShards;
+    }
+
+    /**
+     * Sets the number of shard copies that should be active for index creation to return.
+     * Defaults to {@link ActiveShardCount#DEFAULT}, which will wait for one shard copy
+     * (the primary) to become active. Set this value to {@link ActiveShardCount#ALL} to
+     * wait for all shards (primary and all replicas) to be active before returning.
+     * Otherwise, use {@link ActiveShardCount#from(int)} to set this value to any
+     * non-negative integer, up to the number of copies per shard (number of replicas + 1),
+     * to wait for the desired amount of shard copies to become active before returning.
+     * Index creation will only wait up until the timeout value for the number of shard copies
+     * to be active before returning.  Check {@link CreateIndexResponse#isShardsAcknowledged()} to
+     * determine if the requisite shard copies were all started before returning or timing out.
+     *
+     * @param waitForActiveShards number of active shard copies to wait on
+     */
+    public CreateIndexRequest waitForActiveShards(ActiveShardCount waitForActiveShards) {
```

Comment:
maybe the one taking an int-argument is enough?

Reply:
Then again, I see we still use the ActiveShardCount object in the request converters. Maybe we should kick those out as well, but that would probably be a bit too much for this PR:

Reply:
Maybe we could go for consistency with other HLRC requests like `UnfreezeIndexRequest`, and actually remove the version that takes an `int`?

Reply:
im all for making the client side api easier to deal with. If there is no need for an object, make it just set an int. Even if we set the object in the background, maybe make it an int from the client perspective.

Reply:
I ended up keeping it as an object, so that users are able to use existing constants objects like `ActiveShardCount.ALL` and `ActiveShardCount.DEFAULT`. It also matches what is currently done in other HLRC request classes.